### PR TITLE
Remove obsolete "make test-e2e" guide

### PR DESCRIPTION
**Pull request by cyb70289**

### CODE REVIEW

File: contributors/devel/development.md
Comment (cyb70289) [inline contributors/devel/development.md:469]: This command is invalid now.
```
$ make test-e2e
make: *** No rule to make target 'test-e2e'.  Stop.
```

File: contributors/devel/sig-testing/e2e-tests.md
Comment (cyb70289) [inline contributors/devel/sig-testing/e2e-tests.md:389]: `local-up-cluster.sh` changes owner of client admin key file to `USER` [1], which is supposed to be current non-root user. But we are running the command with `sudo`, the USER will always to root [2], and the current user cannot read client admin key to access the deployed cluster.

[1] https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L628
[2] https://github.com/kubernetes/kubernetes/blob/master/hack/local-up-cluster.sh#L57

File: contributors/devel/development.md
Comment (spiffxp) [reply contributors/devel/development.md:469]: /cc @dims @BenTheElder 
I somehow completely missed that y'all removed this in https://github.com/kubernetes/kubernetes/pull/84696

This seems like a bad developer experience? What do you two do now to run e2e tests against local development?

File: contributors/devel/sig-testing/e2e-tests.md
Comment (spiffxp) [reply contributors/devel/sig-testing/e2e-tests.md:389]: I'd prefer to defer to @BenTheElder and @dims here as well as I don't have much experience with `hack/local-up-cluster.sh`

**Comment by spiffxp** [comment]
Thanks for the pull request! These are valid contributions I just didn't realize that was the state of things today.  I'd like to get enough info to open up a followup issue for the first thing before we merge this.

File: contributors/devel/development.md
Comment (chendave) [reply contributors/devel/development.md:469]: one way to workaround is building `e2e.test` binary, the binary can still work well with local provider, something like this, 
```
/usr/bin/e2e.test -ginkgo.focus=AdmissionWebhook  -provider=local -report-dir=/var/log/kubernetes/e2e
```

File: contributors/devel/sig-testing/e2e-tests.md
Comment (dims) [reply contributors/devel/sig-testing/e2e-tests.md:389]: Agree let's go with this @cyb70289 
```
[dims@dims-a01 21:08] ~/go/src/k8s.io/kubernetes/hack ⟩ rg USER local-up-cluster.sh
57:USER=${USER:-$(whoami)}
628:    ${CONTROLPLANE_SUDO} chown "${USER}" "${CERT_DIR}/client-admin.key" # make readable for kubectl
```

**Comment by cyb70289** [comment]
@BenTheElder might you have time to review this pr?

File: contributors/devel/sig-testing/e2e-tests.md
Comment (BenTheElder) [reply contributors/devel/sig-testing/e2e-tests.md:389]: I sort of thing it should be owned by root given we needed root to run this?
Because this kubeconfig has control over the container runtime on the host outright and is tantamount to being root.

You could make a similar argument about KIND, but it is not such a straightforward control, and if the user has docker socket access limited to a particular user they will also need to run kind as that user and similarly not own the file.

File: contributors/devel/development.md
Comment (BenTheElder) [reply contributors/devel/development.md:469]: Nobody was running this make target? As evidenced by nobody missing it or complaining :-)

e2e tests are run through a _lot_ of different workflows, but hack/e2e.go was terrible.

I personally use the e2e-k8s.sh script from KIND most of the time, but, kubetest, kubetest2, or e2e.test directly are all viable.

File: contributors/devel/sig-testing/e2e-tests.md
Comment (cyb70289) [reply contributors/devel/sig-testing/e2e-tests.md]: Agreed this is a security issue. I've reverted the change.

**Review by BenTheElder — APPROVED** [review]

**Comment by cyb70289** [comment]
@spiffxp @dims, I've updated this PR per @BenTheElder's comment, would you review it again? Thanks.

**Pull request merged**
